packages/clay-navigation/src/Vertical.tsx

```diff
@@ -0,0 +1,143 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import ClayIcon from '@clayui/icon';
+import React from 'react';
+import {Nav} from './Nav';
+import {NavItem} from './NavItem';
+import {NavLink} from './NavLink';
+import {useTransitionHeight} from '@clayui/shared';
+
+interface IItemProps {
+	active?: boolean;
+	onClick?: () => void;
+	href?: string;
+	label?: string;
+}
+
+interface IItem extends IItemProps {
+	initialExpanded?: boolean;
+	items?: Array<IItemProps>;
+}
+
+interface IProps {
+	activeLabel: string;
+	items: Array<IItem>;
+	large?: boolean;
+	spritemap?: string;
+}
+
+interface INavItemProps extends IItem {
+	level: number;
+	spritemap?: string;
+}
+
+function Item({
+	active,
+	href,
+	initialExpanded = false,
+	items: subItems,
+	label,
+	level,
+	spritemap,
+}: INavItemProps) {
+	const menuRef = React.useRef(null);
+	const [expanded, setExpaned] = React.useState(!initialExpanded);
+
+	const [
+		transitioning,
+		handleTransitionEnd,
+		handleClickToggler,
+	] = useTransitionHeight(expanded, setExpaned, menuRef);
+
+	const showIconCollapsed = !(
+		(!expanded && transitioning) ||
+		(expanded && !transitioning)
+	);
+
+	return (
+		<NavItem>
+			<NavLink
+				active={active}
+				collapsed={showIconCollapsed}
+				href={href}
+				onClick={e => {
+					handleClickToggler(e);
+				}}
+				role="button"
+				showIcon={!!subItems}
+				spritemap={spritemap}
+			>
+				{label}
+			</NavLink>
+
+			{subItems && (
+				<div
+					className={classNames({
+						collapse: !transitioning,
+						collapsing: transitioning,
+						show: expanded,
+					})}
+					onTransitionEnd={handleTransitionEnd}
+					ref={menuRef}
+				>
+					<Nav stacked>
+						{renderItems(subItems, spritemap, level++)}
+					</Nav>
+				</div>
+			)}
+		</NavItem>
+	);
+}
+
+function renderItems(items: Array<IItem>, spritemap?: string, level = 0) {
+	return items.map((item, i) => {
+		const key = `${level}-${i}`;
+
+		return <Item {...item} key={key} level={level} spritemap={spritemap} />;
+	});
+}
+
+export const Vertical: React.FunctionComponent<IProps> = ({
+	activeLabel,
+	items,
+	large,
+	spritemap,
+}) => {
+	const [active, setActive] = React.useState(false);
+
+	return (
+		<nav
+			className={classNames('menubar menubar-transparent', {
+				['menubar-vertical-expand-lg']: large,
+				['menubar-vertical-expand-md']: !large,
+			})}
+		>
+			<button
```

Comment:
Just to know, could we use a `ClayButtonWithIcon` here?

Reply:
i forgot about that one. good catch

Reply:
Actually, I just realized we need to use it without `ClayButtonWithIcon` since this component requires `{activeLabel}` to be in the button and next to the icon. `ClayButtonWithIcon` is only for just an icon.